src/setup/validateTxnFields.ts

```diff
@@ -199,12 +202,19 @@ export const validateTxnFields =
         }
         debugAppdata.set(txHash, appData)
 
-        if (isSigned && isSignatureValid) {
+        if (!gasValid) {
+          success = false
```

Comment:
This checks if the signature exists and looks okay but does not do the crypto operation to verify the signature

Reply:
the whole logic of function is broken, cuz reason will be overwritten by further checks, there are no early returns

Reply:
This functions is very inconsistent with  early returns for failures, but these 3 cases here would make sense to return no reason to keep validating if no gas or valid signature